services/core-api/app/api/document_generation/models/document_template.py

```diff
@@ -50,22 +51,34 @@ def _form_spec_with_context(self, primary_key):
             raise Exception('Context Object not found')
 
         for item in spec:
+
+            # Handle "relative data path"
             relative_data_path = item.get('relative-data-path')
-            if not relative_data_path:
-                current_app.logger.debug(f'No relative-data-path for {item["id"]}')
+            if relative_data_path:
+                current_object = source_obj_instance
+                for x in relative_data_path.split('.'):
+                    current_app.logger.debug(f'getting {current_object}.{x}')
+                    current_object = getattr(current_object, x)
+                    if current_object is None:
+                        break
+
+                current_app.logger.info(
```

Comment:
is this worth logging at this level? info will show up in our loki/grafana logs I think?

Reply:
I can change it and the others in this function to debug. They were set at info before I tabbed the code.